File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.py
class Solution:
    def average(self, salary: List[int]) -> float:
        counter, min_sal ,max_sal, sum = 0,float('inf'),float('-inf'),0
        for sal in salary:
            counter+=1
            min_sal = min(min_sal,sal)
            max_sal = max(max_sal,sal)
            sum += sal
        return (sum-min_sal-max_sal)/(counter-2)

File: 1491-average-salary-excluding-the-minimum-and-maximum-salary/1491-average-salary-excluding-the-minimum-and-maximum-salary.java
class Solution {
    public double average(int[] salary) {
        int min_sal = Integer.MAX_VALUE, max_sal = Integer.MIN_VALUE, counter = 0;
		double sum = 0;
		for(int sal : salary){
		    min_sal = Math.min(min_sal, sal);
		    max_sal = Math.max(max_sal, sal);
		    sum += sal;
		    counter++;
		}
		return (sum-min_sal-max_sal)/(counter-2);   
    }
}